### ISSUE TITLE
Index creation via REST API does not apply passed-in `options`

### ISSUE THREAD

**Issue by tatu-at-datastax** [stargate/stargate#2821]
(reported by user)

Looks like `options` `Map` passed via POST to `/v2/schemas/keyspaces/{keyspaceName}/tables/{tableName}/indexes` (and payload bound as `Sgv2IndexAddRequest`) is somehow not being used for index creation.
It is not yet clear where information is dropped: integration test that tries to pass unrecognized options does not fail; `options` make their way to REST API handler at least, and are used for Query creation for request to Bridge.

Original report showed expected matching CQL:

```
create CUSTOM INDEX on route (race_name)
  USING 'org.apache.cassandra.index.sai.StorageAttachedIndex' WITH OPTIONS = { 'case_sensitive': 'false'};
```

for REST payload of:

```json
{
 "column": "race_name",
 "options":
  {
   "case_sensitive": "false"
  }
}
```

**Comment by tatu-at-datastax** [comment]
Looks like the issue is that `QueryBuilder` will only add `options` (`customIndexOptions`) if `customIndexClass` is non-null:

```
    if (customIndexClass != null) {
      query.append(" USING").append(String.format(" '%s'", customIndexClass));
      if (customIndexOptions != null && !customIndexOptions.isEmpty()) {
        query.append(
            customIndexOptions.entrySet().stream()
                .map(e -> String.format("'%s': '%s'", e.getKey(), e.getValue()))
                .collect(Collectors.joining(", ", " WITH OPTIONS = { ", " }")));
      }
    }

```

so probably need to pass explicit customIndexClass to make sure options are added as expected.

**Comment by tatu-at-datastax** [comment]
Fixed via #2822, closing.
